### PR TITLE
bump k8s packages to 1.17.6

### PR DESCRIPTION
**Pull request by dprotaso**
Upstream change knative/pkg#1285

**Release Note**

```release-note
Bumps K8s dependencies to 1.17.5
This makes our minimum K8s version 1.16
```

### CODE REVIEW

**Comment by dprotaso** [comment]
/hold 

I should wait for serving & eventing to merge and then update those deps

File: go.mod [hunk 1]
```diff
@@ -34,24 +34,24 @@ require (
...
-	k8s.io/api v0.17.4
-	k8s.io/apimachinery v0.17.4
-	k8s.io/client-go v12.0.0+incompatible
-	knative.dev/eventing v0.15.1-0.20200526194657-a0991ef60ca9
-	knative.dev/pkg v0.0.0-20200526194649-c48d2ebdae96
-	knative.dev/serving v0.15.1-0.20200526190149-e88e1d6b8eeb
-	knative.dev/test-infra v0.0.0-20200526171149-39b911556285
+	k8s.io/api v0.17.6
+	k8s.io/apimachinery v0.17.6
+	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
```
Comment (grantr) [inline go.mod:39]: Why did this get downgraded?

File: hack/update-k8s-dep.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
```
Comment (grantr) [inline hack/update-k8s-dep.sh:1]: Rename this to update-k8s-dep*s*.sh? Feel free to ignore if this is coming from other repos.

File: go.mod (hunk 1 above)
Comment (dprotaso) [reply go.mod:39]: there’s a replace directive to use the version we want - I believe test-infra has a dep on some k8s repo at this version

File: hack/update-k8s-dep.sh (hunk 2 above)
Comment (dprotaso) [reply hack/update-k8s-dep.sh:1]: Yeah I have it in other repo - I could it in a separate PR

File: go.mod (hunk 1 above)
Comment (dprotaso) [reply go.mod:39]: https://knative.slack.com/archives/C012FCAHN4U/p1588700133341700

**Pull request merged**